### ISSUE TITLE
os/exec: NetBSD kernel crash during os/exec tests

### ISSUE THREAD

**Issue by bradfitz** [golang/go#20836]
Debugging why our NetBSD builders were disappearing, I created a netbsd-amd64-71 instance, enabled serial port support, and watched the serial port while I ran:

```
$ export MOTE=$(gomote create netbsd-amd64-71)
$ gomote run $MOTE go/src/make.bash
$ gomote run $MOTE go/bin/go test -short -v os/exec
```

It works sometimes, hangs sometimes (#20835), and sometimes kernel panics:
```
2017/06/28 19:24:51 [0xc420071b80] Run = ok, after 2.290432897s
2017/06/28 19:25:44 [0xc4203f0c60] Running /tmp/workdir/go/bin/go with args ["/tmp/workdir/go/bin/go" "test" "-short" "-v" "os/exec"] and env ["PWD=/" "HOME=
/" "PATH=/usr/pkg/bin:/usr/pkg/sbin:/sbin:/usr/sbin:/bin:/usr/bin:/usr/pkg/sbin:/usr/pkg/bin:/usr/X11R7/bin:/usr/X11R6/bin:/usr/local/sbin:/usr/local/bin" "_
rc_original_stdout_fd=7" "GOROOT_BOOTSTRAP=/tmp/workdir/go1.4" "_rc_original_stderr_fd=8" "RC_PID=  10" "_rc_pid=2" "_rc_postprocessor_fd=9" "WORKDIR=/tmp/wo
rkdir" "GO_BUILDER_NAME=netbsd-amd64-71"] in dir /tmp/workdir/go/bin
uvm_fault(0xfffffe8071e49b90, 0x0, 1) -> e
fatal page fault in supervisor mode
trap type 6 code 0 rip ffffffff805d185d cs 8 rflags 10286 cr2 8 ilevel 0 rsp fffffe80013f9bf0
curlwp 0xfffffe8059c6f140 pid 5782.5 lowest kstack 0xfffffe80013f72c0
panic: trap
cpu0: Begin traceback...
vpanic() at netbsd:vpanic+0x13c
snprintf() at netbsd:snprintf
startlwp() at netbsd:startlwp
alltraps() at netbsd:alltraps+0x96
mutex_vector_enter() at netbsd:mutex_vector_enter+0x10d
filt_piperead() at netbsd:filt_piperead+0x9d
kevent1() at netbsd:kevent1+0x599
sys___kevent50() at netbsd:sys___kevent50+0x33
syscall() at netbsd:syscall+0x9a
--- syscall (number 435) ---
45db43:
cpu0: End traceback...
dumping to dev 4,1 (offset=8, size=471708):
dump 157 156 155 154 153 152 151 150 149 148 147 146 145 144 143 142 141 140 139 138 137 136 135 134 133 132 131 130 129 128 127 126 125 124 123 122 121 120 
119 118 117 116 115 114 113 112 111 110 109 108 107 106 105 104 103 102 101 100 99 98 97 96 95 94 93 92 91 90 89 88 87 86 85 84 83 82 81 80 79 78 77 76 75 74
 73 72 71 70 69 68 67 66 65 64 63 62 61 60 59 58 57 56 55 54 53 52 51 50 49 48 47 46 45 44 43 42 41 40 39
```

/cc @bsiegert

**Comment by bradfitz** [comment]
A see a number of maybe-related open NetBSD bugs with similar error messages at http://gnats.netbsd.org/summary/category/kern.html:

http://gnats.netbsd.org/44402
http://gnats.netbsd.org/42319
http://gnats.netbsd.org/52065
http://gnats.netbsd.org/46224

And many more.

I'm going to assume this is known.

@bsiegert, as our NetBSD liaison, I'll let you escalate if needed.

**Comment by krytarowski** [comment]
We have several similar issues in the kernel around kevent(2). Current this is a can of bugs... I planned to work on them once I will be done with LLDB porting.... but it will take a while (hopefully till the end of this year). While there there are from time to time fixes in NetBSD-current and they are in NetBSD 8_BETA.

**Comment by bradfitz** [comment]
@krytarowski, thanks for the update.

**Comment by mikioh** [comment]
See #14127.

**Comment by bradfitz** [comment]
Fixed by requiring NetBSD 8+ for Go 1.10.

Documentation bug is #22911
